pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -158,27 +162,33 @@ func getPackagesFromCB(ctx context.Context, clusterBootstrap *runtanzuv1alpha3.C
 			if pkgShortName == "tkg-storageclass" {
 				continue
 			}
-			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: constants.TkgNamespace},
+			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: systemNamespace},
 				Spec: kapppkgv1alpha1.PackageSpec{RefName: pkgName, Version: pkgVersion}})
 		}
 	}
 
 	return packages, nil
 }
 
-// checkClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
-func checkClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
+// CheckClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
+func CheckClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
 	log.Infof("Verify addons on workload cluster %s with management cluster %s", wcClusterName, mcClusterName)
 
 	var clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap
+
+	systemNamespace := constants.TkgNamespace
+	if infrastructureName == "tkgs" {
+		systemNamespace = constants.TKGSClusterClassNamespace
+	}
+
 	if isManagementCluster {
 		clusterBootstrap = getClusterBootstrap(ctx, mccl, mcClusterNamespace, mcClusterName)
```

Comment:
As there is no CB for supervisor cluster, here in line 184 you need to also do this check as well, so that fetching CB for the supervisor cluster is not attempted.
```
if isManagementCluster && infrastructureName != "tkgs" {..}
```

Reply:
Given CB only exists on supervisor cluster in workload cluster namespace for tkgs, from the test file itself we are not calling this for management cluster. Only the else is exercised.

Reply:
I see. If we are sure that to-be-added BYOT tests also don't ever run these on supervisor cluster, then we are good.